AirCastingTests/Logic/SDSync/SdSyncAveragingServiceTests.swift

```diff
@@ -0,0 +1,135 @@
+        
+        var measurementsInInterval: [SDSyncMeasurement] = []
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: startDate, averagingWindow: .firstThresholdWindow, action: { _, measurements in measurementsInInterval = measurements })
+        
+        XCTAssertEqual(reminder, [SDSyncMeasurement(measuredAt: startDate+TimeInterval(thresholdWindow.rawValue), value: 3)])
+        XCTAssertEqual(measurementsInInterval, [SDSyncMeasurement(measuredAt: startDate, value: 2), SDSyncMeasurement(measuredAt: startDate+1, value: 1)])
+    }
+    
+    func test_averageMeasurementsWithReminderCalledForFirstThresholdWindow_calculatesRightAverages() throws {
+        let thresholdWindow = AveragingWindow.firstThresholdWindow
+        let numberOfIntervals = 6
+        let startDate = Date.init(timeInterval: TimeInterval(-thresholdWindow.rawValue * numberOfIntervals), since: Date())
+        
+        var measurements: [SDSyncMeasurement] = []
+        var expectedAverages: [Double] = []
+        var valuesForAveraging: [Double] = []
+        
+        for i in 0...numberOfIntervals*thresholdWindow.rawValue {
+            if i != 0 && i%thresholdWindow.rawValue == 0 {
+                expectedAverages.append(valuesForAveraging.reduce(0.0, +) / Double(valuesForAveraging.count))
+                valuesForAveraging = []
+            }
+            valuesForAveraging.append(Double(i))
+            measurements.append(SDSyncMeasurement(measuredAt: startDate + TimeInterval(i), value: Double(i)))
+        }
+        
+        var resultMeasurements: [SDSyncMeasurement] = []
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: startDate, averagingWindow: thresholdWindow, action: { measurement, _ in resultMeasurements.append(measurement) })
+        
+        XCTAssertEqual(resultMeasurements.map(\.value), expectedAverages)
+    }
+    
+    func test_averageMeasurementsWithReminderCalledForSecondWindow_calculatesRightAverages() throws {
+        let thresholdWindow = AveragingWindow.secondThresholdWindow
+        let numberOfIntervals = 6
+        let startDate = Date.init(timeInterval: TimeInterval(-thresholdWindow.rawValue * numberOfIntervals), since: Date())
+        
+        var measurements: [SDSyncMeasurement] = []
+        var expectedAverages: [Double] = []
+        var valuesForAveraging: [Double] = []
+        
+        for i in 0...numberOfIntervals*thresholdWindow.rawValue {
+            if i != 0 && i%thresholdWindow.rawValue == 0 {
+                expectedAverages.append(valuesForAveraging.reduce(0.0, +) / Double(valuesForAveraging.count))
+                valuesForAveraging = []
+            }
+            valuesForAveraging.append(Double(i))
+            measurements.append(SDSyncMeasurement(measuredAt: startDate + TimeInterval(i), value: Double(i)))
+        }
```

Comment:
Maybe we could change this `i` to `newValue` of something that could indicate broader context?